aten/src/ATen/cuda/CUDAEvent.h

```diff
@@ -45,6 +46,10 @@ struct TORCH_CUDA_CPP_API CUDAEvent {
     try {
       if (is_created_) {
         CUDAGuard guard(device_index_);
+        const c10::impl::PyInterpreter* interp = c10::impl::CUDATrace::get_trace();
+        if (C10_UNLIKELY(interp)) {
+          interp->trace_cuda_event_deletion(reinterpret_cast<uintptr_t>(event_));
```

Comment:
I know this code is originally from me, but it occurs to me that if we can assume these are pointer sized, we could probably just cast to `void*` or even just forward declare the CUDA types inside c10 non-cuda so we can reference them (this is what we do for PyObject from Python, so that we can say PyObject inside non-Python headers). Not important enough to block this PR though IMO.